### PR TITLE
Fix signatory bugs in develop

### PR DESCRIPTION
**Pull request by cimadai**
## What is this pull request?
Fix AddSignatory / RemoveSignatory command's bug.

## Why do you implement it? Why do we need this pull request?

iroha has following bugs.

### What is this issue?

If you want to add signatory to a user, iroha causes following postgres error.
    > insert or update on table "account_has_signatory" violates foreign key constraint "account_has_signatory_public_key_fkey".
Because not inserted to signatory table before insertion to account_has_signatory table.

### How resolved?

I inserted insertSignatory method before insertAccountSignatory method.

### What is this issue?

If you want to remove a signatory from a user, but the signatory remains in signatory table.
Because not deleted the signatory from signatory table after deletion from account_has_signatory table.

### How resolved?

I added deleteSignatory method and inserted the method after deleteAccountSignatory method.

### CODE REVIEW

**Comment by cimadai** [comment]
Thank you. I added unit test.

**Comment by neewy** [comment]
@cimadai thank you for your contribution. We will rebase&merge it ASAP

File: irohad/model/execution/impl/command_executor.cpp [hunk 1]
```diff
@@ -331,7 +332,8 @@ bool RemoveSignatoryExecutor::execute(const Command &command,
 
   // Delete will fail if account signatory doesn't exist
   return commands.deleteAccountSignatory(remove_signatory.account_id,
-                                         remove_signatory.pubkey);
+                                         remove_signatory.pubkey) &&
+         commands.deleteSignatory(remove_signatory.pubkey);
```
Comment (neewy) [inline irohad/model/execution/impl/command_executor.cpp]: it can fail in case if other account has this signatory

**Comment by cimadai** [comment]
@neewy I fixed it and added some test.
please check it.

File: irohad/ametsuchi/impl/postgres_wsv_command.cpp [hunk 2]
```diff
@@ -105,10 +105,9 @@ namespace iroha {
       try {
         pqxx::binarystring public_key(signatory.data(), signatory.size());
         transaction_.exec(
-            "INSERT INTO signatory(\n"
-            "            public_key)\n"
-            "    VALUES (" +
-            transaction_.quote(public_key) + ");");
+            "INSERT INTO signatory(public_key)\n"
```
Comment (l4l) [inline irohad/ametsuchi/impl/postgres_wsv_command.cpp]: What's the purpose of newline char?

Comment (cimadai) [reply irohad/ametsuchi/impl/postgres_wsv_command.cpp]: @l4l Thanks your comment!
There is no special reason.
Is it something strange?

**Comment by cimadai** [comment]
@lebdron Thanks! Added it and some tests.

**Comment by cimadai** [comment]
fixed conflict.

File: irohad/ametsuchi/impl/postgres_wsv_command.cpp (hunk 2 above)
Comment (l4l) [reply irohad/ametsuchi/impl/postgres_wsv_command.cpp]: Nope, just curious :)

**Comment by lebdron** [comment]
@cimadai Is it possible to add it exactly to `RemoveSignatoryExecutor::isValid` method? Adding it to both `RemoveSignatoryExecutor::execute` and previous method seems redundant.
Also, is it possible to make many short simple test for each case instead of adding everything to `AddSignatoryTest`? That seems to be more readable.

**Comment by neewy** [comment]
@cimadai Shimada-san, have you checked comment of @lebdron? What do you think about that?

**Comment by cimadai** [comment]
@lebdron @neewy OK, I'll change it.

**Comment by neewy** [comment]
@cimadai Shimada-san, hi! Is there any progress on the suggestion of @lebdron ?

**Pull request closed without merging**